### ISSUE TITLE
Get Albany to compile with -DENABLE_ENSEMBLE:BOOL=ON

### ISSUE THREAD

**Issue by ikalash** [gahansen/Albany#42]
This has been broken for awhile.

**Comment by ibaned** [comment]
@agsalin is this something we want to spend time on and preserve ? we may want to talk with @etphipp about the way to move forward on this (there was some discussion on our side about storing Ensemble types directly as the scalar type in Tpetra objects instead of converting to multiple Vectors etc).

**Comment by agsalin** [comment]
Yes, we need to continue our talk with Eric about a path forward. We need to balance our desire for more capabilities with the maintainability/extensibility of the code. I'm still not clear on how much work it will be to get the ensemble capability back through the data type, and whether that work is slated to happen under ATDM or if Albany would be the driver.

**Comment by jfike** [comment]
I had started down the Tpetra path, which involves templating the workset and AlbanyTraits definitions.  I made some progress down this path, but have not found the time to finish it so far.

As far as the current issue with building with -D ENABLE_ENSEMBLE = ON, I started on an attempt to fix it a while ago, but have not had the time to resolve the issue yet.  Hopefully I can find the time to do so fairly soon.
